jdbc/src/main/clojure/xtdb/next/jdbc.clj

```diff
@@ -47,6 +58,16 @@
 
 (def col-reader (->col-reader :kebab-case-keyword))
 
+
+(def ^{:doc "Builder function which converts result-set rows' map keys to kebab-cased keywords (iteratively)"}
+  builder-fn
+  (nj-rs/as-maps-adapter
+   (fn [rs opts]
+     (nj-rs/as-unqualified-modified-maps
```

Comment:
let's stick with the one from optional (as the test-util was) - we're looking to round-trip documents as they were inserted, and I think I'd want to err on the side of dropping null values than inserting them (although it's obviously a subjective decision).